ReactAndroid/src/main/java/com/facebook/react/views/text/ReactInlineBorderSpan.java

```diff
@@ -0,0 +1,105 @@
+ */
+public class ReactInlineBorderSpan implements LineBackgroundSpan, ReactSpan {
+  private final int effectiveStart;
+  private final int effectiveEnd;
+  private final int effectiveFontSize;
+
+  private final int backgroundColor = Color.parseColor("#f6f6f6");
+  private final int borderColor = Color.parseColor("#b3b8bc");
+  private final int borderRadius = 8;
+  private final int borderWidth = 2;
+
+  public ReactInlineBorderSpan(int effectiveFontSize, int effectiveStart, int effectiveEnd) {
+    this.effectiveFontSize = effectiveFontSize;
+    this.effectiveStart = effectiveStart;
+    this.effectiveEnd = effectiveEnd;
+  }
+
+  /**
+   * Calculates text start and end position per given lines text range
+   */
+  private int[] calculateBorderedTextCharRange(int lineStart, int lineEnd) {
+    /**
+     * If bordered text's position falls in lines size range return the position of bordered text's first char
+     * Otherwise return lines start position.
+     * 
+     * e.g. given paragraph below:
+     * This is a sample text [and the bordered | prependedTextStart -> 23, prependedTextEnd -> 39
+     * text starts and ends here].             | prependedTextStart -> 24, prependedTextEnd -> 65
+     */
+    int prependedTextStart = (lineStart < this.effectiveStart && lineEnd > this.effectiveStart) ? this.effectiveStart : lineStart;
+    int prependedTextEnd = (lineEnd < effectiveEnd) ? lineEnd : effectiveEnd;
+
+    return new int[]{prependedTextStart, prependedTextEnd};
+  }
+
+  /**
+   * Generate Rect that will be used to cover bordered background layer.
+   */
+  private RectF generateTextBounds(Paint paint, int top, int bottom, CharSequence text, int start, int end) {
+    int[] borderedTextRange = this.calculateBorderedTextCharRange(start, end);
+    
+    /**
+     * Calculate bordered text's background layer position relative to the line.
+     * 
+     * e.g. given the line below
+     * This is a prepended [this is bordered] text.
+     */
+    int prependedTextWidth = Math.round(paint.measureText(text, start, borderedTextRange[0]));
+    int borderedTextWidth = Math.round(paint.measureText(text, borderedTextRange[0], borderedTextRange[1]));
+
+    RectF rect = new RectF();
+    int leftPosition = prependedTextWidth;
+    int rightPosition = prependedTextWidth + borderedTextWidth;
+    rect.set(prependedTextWidth, top, prependedTextWidth + borderedTextWidth, bottom);
+
+    return rect;
+  }
+
+  @Override
+  public void drawBackground(Canvas canvas, Paint paint, int left, int right, int top, int baseline, int bottom, CharSequence text, int start, int end, int lineNumber) {
+    /**
+     * Set's RN Text size for the canvas to measure accurate metrics
+     */
+    paint.setTextSize(this.effectiveFontSize);
+
+    RectF rect = generateTextBounds(paint, top, bottom, text, start, end);
+
+    /**
+     * Draw filled background 
+     */
+    Paint backgroundPaint = new Paint();
+    backgroundPaint.setColor(this.backgroundColor);
```

Comment:
Can we be certain that this.backgroundColor won't return null?

Reply:
Same for `borderWidth, borderRadius, and borderPaint` below

Reply:
You are right, added an extra check in the constructor.

`ReactInlineBorderSpan` class won't be applied if `textCodeBlock` prop is not provided to the Text component. And null values are handled by the setColor, setStrokeWidth ... Therefore I'm just thinking if it actually makes sense to micro-manage this further with multiple if/elses at `drawBackground` method ?